src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -417,6 +422,61 @@ public void AddHostObject(string name, FormulaType type, Func<IServiceProvider,
                 displayName: default);
 
             _variables.Add(hostDName, info);
-        }
+        }
+
+        /// <summary>
+        /// Adds a named type that can be referenced in expression.
+        /// </summary>
+        /// <param name="typeName">Name of the type to be added into Symbol table.</param>
+        /// <param name="type">Type associated with the name.</param>
+        public void AddType(DName typeName, FormulaType type)
+        {
+            Contracts.AssertValue(typeName.Value);
+            Contracts.AssertValue(type);
+            Contracts.Assert(typeName.Value.Length > 0);
+
+            using var guard = _guard.Enter(); // Region is single threaded.
+            Inc();
+
+            _namedTypes.Add(typeName, type);
+        }
+
+        public void AddTypes(IEnumerable<KeyValuePair<DName, FormulaType>> types)
+        {
+            Contracts.AssertValue(types);

```

Comment:
> Contracts.AssertValue(types); [](http://example.com/codeflow?start=11&length=30)

 Contracts.AssertValue(types);
 Contracts.AssertAllValues(types);

Reply:
resolving since keyvalue pair is struct